dashboard/modules/reporter/reporter_agent.py

```diff
@@ -605,9 +605,12 @@ def _get_workers(self):
         if raylet_proc is None:
             return []
         else:
-            workers = {
-                self._generate_worker_key(proc): proc for proc in raylet_proc.children()
-            }
+            workers = {}
+            # windows, get the child process not the runner
+            for child in raylet_proc.children():
+                if child.children():
+                    child = child.children()[0]
```

Comment:
I think we should do it only for windows, otherwise if a Ray task launches a child process, this will find that process not the Ray task worker process.

Reply:
Good point, that would happen on windows too. I wonder if there is some better indicator that the process is the one we are looking for. I should add a test for a worker that has `subprocess.Popen()` in its task.

Reply:
It should be fine since we only get child of child and we know the direct child must be a launcher?